### DISCUSSION TITLE
Is there a way to have public and non-public repos distinguish configuration, without using multiple presets?

### DISCUSSION THREAD

**Discussion by jamietanna** [renovatebot/renovate#29746] [Request Help] [answered]
### What would you like help with?

I would like help with my configuration

### How are you running Renovate?

Mend Renovate hosted app on github.com

### Please tell us more about your question or problem

We're looking to have slightly different configuration between public and non-public repositories, for instance to tune some of the specific labels or information that's made visible for a non-public repo.

This would be managed through a central base configuration, such as `:base`.

We were hoping to be able to have this `:base` configuration tune its configuration based on whether it's targeting a public or a non-public repository (on GitHub).

This may be under the `packageRules`, but that probably wouldn't make sense for some of the other configuration we'd want.

Is this possible to do within Renovate right now? I presume it's not, and may not be something we want to add to Renovate and expose to our users, but thought I'd ask.

The way we're looking at solving it in the immediate term is to have two presets:

```json
{
    "description": "Base config",
    "labels": [
        "dependencies"
    ]
}
```

```json
{
    "description": "Non-public config",
    "labels": [
        "dependencies",
        "team-xyz"
    ]
}
```

(a contrived example)

That way, all repos can inherit `:base`, and then a repository that isn't public can also inherit `:non-public`.

### Logs (if relevant)

<details><summary>Logs</summary>

```

Replace this text with your logs, between the starting and ending triple backticks

```

</details>

**Comment by viceice** [comment, marked as answer]
is not possible and will probably never be possible via package rules. the required information isn't available.

so use two presets as solution

**Reply by rarkins** [reply]
I was thinking that we should migrate from private/public repo concepts to "visibility" public/private (and I think "internal" is an additional one for GitHub). Then we could add a matchVisibility option to packageRules

**Reply by rarkins** [reply]
On second look I don't see the terms private/public in platform code, so maybe I misremembered.

I think we could do this:

1. Add `visibility` field to `RepoResult`: https://github.com/renovatebot/renovate/blob/48e6aa4f8477f81895ca6eed22b39d6cfc3ccd32/lib/modules/platform/types.ts#L32-L36
2. Add new `matchVisibility` option to `packageRules`

**Reply by HonkingGoose** [reply]
Tracking this idea with issue:

- #29871
